File: Assets/zfoolua/common/Heartbeat.lua

local Heartbeat = {}

function Heartbeat:new()
    local obj = {
        
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function Heartbeat:protocolId()
    return 102
end

function Heartbeat:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
end

function Heartbeat:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = Heartbeat:new()
    
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return Heartbeat

File: Assets/zfoocs/Common/Heartbeat.cs
using System;
using System.Collections.Generic;

namespace zfoocs
{
    
    public class Heartbeat
    {
        

        public static Heartbeat ValueOf()
        {
            var packet = new Heartbeat();
            
            return packet;
        }
    }


    public class HeartbeatRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 102;
        }

        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            Heartbeat message = (Heartbeat) packet;
            buffer.WriteInt(-1);
        }

        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            Heartbeat packet = new Heartbeat();
            
            if (length > 0) {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}
